cas-server-support-ldap/src/test/java/org/jasig/cas/authentication/principal/PersonDirectoryPrincipalResolverLdaptiveTests.java

```diff
@@ -0,0 +1,48 @@
+package org.jasig.cas.authentication.principal;
+
+import org.jasig.cas.adaptors.ldap.AbstractLdapTests;
+import org.jasig.cas.authentication.UsernamePasswordCredential;
+import org.jasig.services.persondir.support.ldap.LdaptivePersonAttributeDao;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ldaptive.LdapEntry;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.junit.Assert.*;
+
+/**
+ * This is {@link PersonDirectoryPrincipalResolverLdaptiveTests}.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration({"/ldap-context.xml", "/resolver-context.xml"})
+public class PersonDirectoryPrincipalResolverLdaptiveTests extends AbstractLdapTests {
```

Comment:
This needs an explanation:

1. I realized that Person Directory has a dependency conflict with CAS with its version of Spring. 
2. Once I updated person directory use the latest version of Spring similar to CAS, I noticed tests started to fail. 
3. So What you see here is the result of me fixing person directory and testing it on the CAS side to ensure attributes are rightly resolved.
4. PD is now updated and released as 1.7.1.